chain/actions.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import "github.com/ava-labs/hypersdk/codec"
+
+func getActionSize(action Action) (int, error) {
+	if actionWithSize, ok := action.(HasSize); ok {
+		return actionWithSize.Size(), nil
+	}
+	return codec.LinearCodec.Size(action)
+}
+
+func marshalActionInto(action Action, p *codec.Packer) error {
+	if actionWithMarshal, ok := action.(HasMarshal); ok {
+		actionWithMarshal.Marshal(p)
+		return nil
+	}
+	return codec.LinearCodec.MarshalInto(action, p.Packer)
+}
```

Comment:
Since we are just performing an interface check and then calling the optional interface here, could we switch this to taking in an interface here?

Reply:
No, we can't, because some actions (probably most of them) are not designed to implement the Marshaler interface.

Reply:
The point of this function is to hide the interface check

Reply:
Got it! Fixed, thanks!